### PR TITLE
adding support for terminal buffers

### PR DESCRIPTION
**Pull request by vito-c**
Added support for terminal buffers in neovim

### CODE REVIEW

**Comment by Shougo** [comment]
Why this feature is needed?
Please describe it.

**Comment by vito-c** [comment]
This feature will let you display only terminal buffers when you are using neovim. It is very helpful for my workflow when I have a lot of terminal buffers running processes and I want to jump to one quickly. I don't want to display any of the file buffers.

**Comment by Shougo** [comment]
You can filter the buffers by `:Unite buffer -input=term:`.
Why this feature is needed?

**Comment by vito-c** [comment]
`:Unite buffer -input=term` that does not filter by buffer type. The feature I added will filter the buffer list by type.

**Comment by Shougo** [comment]
OK.   I will review it.

File: autoload/unite/sources/buffer.vim [hunk 1]
```diff
@@ -53,11 +53,14 @@ function! s:source_buffer_all.hooks.on_init(args, context) "{{{
         \ (get(a:args, 0, '') ==# '+')
   let a:context.source__is_minus =
         \ (get(a:args, 0, '') ==# '-')
+  let a:context.source__is_t =
```
Comment (Shougo) [inline autoload/unite/sources/buffer.vim]: Please use `source__is_terminal` instead of `source__is_t`.

File: autoload/unite/sources/buffer.vim [hunk 2]
```diff
@@ -218,7 +223,7 @@ function! s:compare(candidate_a, candidate_b) "{{{
       \ (a:candidate_b.action__buffer_nr == unite#get_current_unite().prev_bufnr ? -1 :
       \ a:candidate_b.source__time - a:candidate_a.source__time)
 endfunction"}}}
-function! s:get_buffer_list(is_bang, is_question, is_plus, is_minus) "{{{
+function! s:get_buffer_list(is_bang, is_question, is_plus, is_minus, is_t) "{{{
```
Comment (Shougo) [inline autoload/unite/sources/buffer.vim]: Please use `is_terminal` instead of `is_t`.

File: autoload/unite/sources/buffer.vim [hunk 3]
```diff
@@ -251,15 +256,17 @@ function! s:get_buffer_list(is_bang, is_question, is_plus, is_minus) "{{{
...
   return bufexists(a:bufnr) &&
         \ (a:is_question ? !buflisted(a:bufnr) :
         \    (a:is_bang || buflisted(a:bufnr)))
         \ && (!a:is_plus || getbufvar(a:bufnr, '&mod'))
-        \ && (!a:is_minus || (getbufvar(a:bufnr, '&buftype') == ''
+        \ && (!a:is_minus || buftype == ''
         \                     && bufname != ''
-        \                     && !isdirectory(bufname)))
+        \                     && !isdirectory(bufname))
+        \ && (!a:is_t || buftype == 'terminal' )
```
Comment (Shougo) [inline autoload/unite/sources/buffer.vim]: Please use `==#` instead of `==`.

**Pull request closed without merging**